CARDS-2510: Fix computed questions always recalculating on checkin

acrowthe opened the pull request:
- Fix computed questions being recalculated by the backend whenever the front end submits the a form with no relevant changed answers

To ensure that computed questions are working as expected:
- Launch in runmode test with --dev enables
- Sign in to user: `computedtestuser` password: `testpassword`
- Create a Backend Computed Question form but do not input any answers or save it
- Check the created form using composum. It should have 2 sections, one with 4 answers and one with 8 answers.
  - Check the section with 8 answers: Two should have strings describing the (lack of) values in Question 4
![image](https://github.com/data-team-uhn/cards/assets/20056751/9c94b6a1-d8f6-4ce1-8d19-844dd4b30171)

- Save the form with no edited answers
- Create a new Backend Computed Question form, fill out all answers and save it.
- Sign out of `computedtestuser` and sign into `admin`
- View the created forms in view mode only and verify that the answers are as expected
  
  
To ensure that the computed questions are only being recalculated when desired:
- Launch cards in runmode test with --dev and a debugger attached
- Put breakpoints on `ComputedAnswersEditor.computeAnswers` on lines 194 and 199 as shown below:
![image](https://github.com/data-team-uhn/cards/assets/20056751/59067944-6fe0-4dcc-80c9-1d9ff1f87348)
- Verify that the line 194 breakpoint is being hit whenever the computation shouldn't be saved
- Verify that the line 199 breakpoint is being hit whenever the computation should be saved:
  - When a form is created all computed questions should be calculated
  - When the question does not take any other question inputs (seen by 1 question always computing with `result=1`)
  - When the form is modified by a user that does not have access to the computed questions and modifies an input answer

acrowthe committed:
Computed questions recalculate on checkin
- Fix computed questions being recalculated by the backend whenever the front end submits the a form with no relevant changed answers

sdumitriu commented on modules/data-model/forms/impl/src/main/java/io/uhndata/cards/forms/internal/ComputedAnswersEditor.java:
```diff
@@ -154,35 +166,57 @@ protected void handleLeave(final NodeState form)
+                answersByQuestionName, resultType, changedQuestions);
+            // Do not compute the question if all the following are true
+            // - The expression depends on other questions
+            // - One of the other questions has changed
```
Am I reading it wrong, or does the code test the opposite, none of the questions were changed?

acrowthe replied:
Comment is incorrect, thanks for catching that

sdumitriu commented on modules/data-model/forms/impl/src/main/java/io/uhndata/cards/forms/internal/ComputedAnswersEditor.java:
```diff
@@ -154,35 +166,57 @@ protected void handleLeave(final NodeState form)
+                answer.setProperty(FormUtils.VALUE_PROPERTY, result, untypedResultType);
+                changedQuestions.add(question.getName());
+            }
+            // Update the computed value in the map of existing answers
```
I wonder if we should also update `changedQuestions`, since the question did change as a result of the computation.

acrowthe committed:
CARDS-2510: Fix computed questions always recalculating on checkin
- Change incorrect code comment
- Fix a case where a changed computed answer wouldn't be marked as changed
